sdv/constraints/tabular.py

```diff
@@ -1139,6 +1273,24 @@ def __init__(self, column_names):
         self.column_names = column_names
         self.constraint_columns = tuple(self.column_names)
 
+    @classmethod
+    def _validate_metadata_specific_to_constraint(cls, metadata, **kwargs):
+        keys = []
+        column_names = kwargs.get('column_names')
+        if isinstance(metadata._primary_key, tuple):
+            primary_key = metadata._primary_key
+        else:
+            primary_key = tuple(metadata._primary_key)
+
+        for column in column_names:
+            if column in metadata._alternate_keys or column in primary_key:
+                keys.append(column)
+
+        raise ConstraintMetadataError(
+            f"A Unique constraint is being applied to columns '{keys}'. "
+            'These columns are already a key for that table.'
+        )
```

Comment:
Shouldn't this be raised in case that there are `keys` ?

Reply:
Isn't that what this is doing?

Reply:
Yup, I forgot to test this one too